File: 10862.cpp
//UVA-10862     Connect the Cable Wires 
//#include <bits/stdc++.h>
#include <algorithm>
#include <cstdio>
#include <cstring>
#include <string>
#include <vector>
#include <queue>
#include <stack>
#include <iostream>
#include <map>
#include <queue>
#include <math.h>
#define INF         1e9
#define LIM         2010
#define UNVISITED   -1
#define loop(i,a,b) for(i=a;i<b;i++)
#define X first
#define Y second
using namespace std;
typedef pair<int,int> ii;

int f[LIM];
  
void fib(){
  f[1] = f[2] = 1;
  for (int i = 2; i < LIM; i++)
      f[i] = f[i-1] + f[i-2];
}

int main(){
  fib();
  int n;
  while(scanf("%d",&n)==1 && n>0){
    printf("%d\n",f[2*n]);
  }
  return 0;
}

File: 10862.java
//UVA-10862 		Connect the Cable Wires 
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class Main {
	private static BigInteger[] f;
	
	private static void fib(){
		f[1] = f[2] = new BigInteger(1+"");
		for (int i = 3; i < 4010; i++)
			f[i] = f[i-1].add(f[i-2]);
	}
	
	public static void main(String[] args) throws Exception {
		f=new BigInteger[4010];
		fib();
		int num;
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		num=Integer.parseInt(br.readLine().trim());
		while(num>0){
			System.out.println(f[2*num]);
			num=Integer.parseInt(br.readLine().trim());
		}
	}
}

